api_docs/rest-error-handling.md

```diff
@@ -44,3 +44,43 @@ to a given request, the values returned will be for the strictest
 limit.
 
 [rate-limiting-rules]: https://zulip.readthedocs.io/en/latest/production/security-model.html#rate-limiting
+
+## Ignored Parameters
+
+All Zulip REST API endpoints may return an array of parameters sent
+in the request that are not supported by that specific endpoint.
+
+While this can be expected, e.g. when sending both current and legacy
+names for a parameter to a Zulip server of unknown version, this often
+indicates either a bug in the client implementation or an attempt to
+configure a new feature while connected to an older Zulip server that
+does not support said feature.
+
+**Changes**:
+
+* Added to all REST API endpoint JSON success responses in Zulip 7.0
+  (feature level 165)
+
+* Added to
+  [`POST /users/me/subscriptions/properties`](/api/update-subscription-settings)
+  in Zulip 5.0 (feature level 111).
+
+* Added to
+  [`PATCH /realm/user_settings_defaults`](/api/update-realm-user-settings-defaults)
+  in Zulip 5.0 (feature level 96).
+
+* Introduced in [`PATCH /settings`](/api/update-settings) in Zulip 5.0
+  (feature level 78).
+
+### Example JSON success response with ignored parameters
+
+```json
+{
+  "ignored_parameters_unsupported": [
+      "invalid_param_1",
+      "invalid_param_2"
+    ],
+  "msg": "",
+  "result": "success",
+}
+```
```

Comment:
We've had bad experiences in the past with examples hardcoded in documentation having typos. While this one is pretty short and thus this isn't worth much time to address, it'd be kinda nice to replace it with something that's verified as matching the actual API.

I can't remember if there's an easy way to include a sample response from `zulip.yaml` that is checked against actual API responses by the test suite, but if you happen to know how to do that, I think it'd be an improvement for this to be a `generate_code_example` sort of thing, that might be more robust.